tests/base.py

```diff
@@ -109,6 +109,7 @@ class AppTestCase(unittest.TestCase):
 
     def setUp(self):
         super(AppTestCase, self).setUp()
+        test_app.config['PRESERVE_CONTEXT_ON_EXCEPTION'] = False
```

Comment:
What does this do in context of the PR?

Reply:
I ran into issues, where flask was raising two exceptions hence confusing the request context. The fix is suggested here [assertionerror-popped-wrong-request-context](https://stackoverflow.com/questions/26647032/py-test-to-test-flask-register-assertionerror-popped-wrong-request-context/28139033#28139033). Travis log for the error [here](https://travis-ci.org/fabmiz/osf.io/jobs/272465037#L2309). Will investigate further.

Reply:
As long as it doesn't cause problems for other tests. If it's possible to put that into the test that was having problems, rather than in the setup, that would be better, but I can see how it might be too late then.